pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -316,6 +322,144 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
+// addLabelController adds a new Controller managing labels to mgr
+func addLabelController(mgr manager.Manager, mutatingClient corev1client.CoreV1Interface) error {
+	labelReconciler := &ReconcileSecretMissingLabel{
+		cachedClient:   mgr.GetClient(),
+		mutatingClient: mutatingClient,
+	}
+	labelController, err := controller.New(labelControllerName, mgr, controller.Options{
+		Reconciler: labelReconciler,
+	})
+	if err != nil {
+		return err
+	}
+
+	missingLabelSecretsMapFn := handler.EnqueueRequestsFromMapFunc(func(_ context.Context, a client.Object) []reconcile.Request {
+		return []reconcile.Request{
+			{
+				NamespacedName: types.NamespacedName{
+					Name:      a.GetName(),
+					Namespace: a.GetNamespace(),
+				},
+			},
+		}
+	})
+
+	missingLabelCredSecretPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return IsMissingSecretLabel(e.ObjectNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return IsMissingSecretLabel(e.Object)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return IsMissingSecretLabel(e.Object)
+		},
+	}
+	// Watch Secrets and reconcile if we see an event for an admin credential secret in kube-system.
```

Comment:
i thought we were specifically *not* going to reconcile(add labels to) the admin secret?

Reply:
Oops, bad copy-pasta